### PR TITLE
Change scaffolds to index on MRN

### PR DESCRIPTION
**Pull request by TheNickWilson**
Users may need to pause work on one arrival notification and work on others in the meantime.  To facilitate this, we key the data on MRN instead of the user's Id; in turn, this means we include the MRN in the URI of almost all pages in the service.  This PR updates the scaffolds to make this the default.

We created `MovementReferenceNumber` as a model to allow for future extension; it currently allows any string to be its value, but we can enrich this in future to validate the format and check digit.

### CODE REVIEW

File: app/models/MovementReferenceNumber.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+package models
+
+import play.api.libs.json._
+import play.api.mvc.PathBindable
+
+final case class MovementReferenceNumber (value: String)
+
+object MovementReferenceNumber {
+
+  implicit lazy val pathBindable: PathBindable[MovementReferenceNumber] = new PathBindable[MovementReferenceNumber] {
```
Comment (amaalali) [inline app/models/MovementReferenceNumber.scala]: 👍

File: conf/app.routes [hunk 2]
```diff
@@ -8,13 +8,13 @@ GET        /language/:lang                              controllers.LanguageSwit
 
 GET        /this-service-has-been-reset                 controllers.SessionExpiredController.onPageLoad
 
-GET        /check-your-answers                          controllers.CheckYourAnswersController.onPageLoad
+GET        /:mrn/check-your-answers                          controllers.CheckYourAnswersController.onPageLoad(mrn: MovementReferenceNumber)
```
Comment (amaalali) [inline conf/app.routes]: 👍

File: test/generators/UserAnswersGenerator.scala [hunk 3]
```diff
@@ -36,7 +36,7 @@ trait UserAnswersGenerator extends TryValues {
 
     Arbitrary {
       for {
-        id      <- nonEmptyString
+        id      <- nonEmptyString.map(MovementReferenceNumber(_))
```
Comment (amaalali) [inline test/generators/UserAnswersGenerator.scala]: We can use the `Arbitrary[MovementReferenceNumber]`
```suggestion
        id      <- arbitrary[MovementReferenceNumber]
```

Comment (TheNickWilson) [reply test/generators/UserAnswersGenerator.scala]: Good spot, thanks :+1:

**Pull request merged**